source/System.cpp

```diff
@@ -240,7 +281,30 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 					}
 			}
 			else
-				fleets.emplace_back(fleet, child.Value(valueIndex + 1));
+			{
+				int period = 200;
+				int limit = LimitedEvents<Fleet>::NO_LIMIT;
+				int initialCount = 0;
+				string id;
+				bool ignoreEnemyStrength = false;
+
+				if(child.Size() > valueIndex + 1)
+					period = child.Value(valueIndex + 1);
+				if(child.HasChildren())
+					LoadFleet(child, period, limit, initialCount, id, ignoreEnemyStrength);
```

Comment:
instead of giving every single variable as parameters to the function, how about giving it the fleet to write the data into?

Reply:
The LimitedEvents and RandomEvent classes have only const members. I could change that.